src/librustc/mir/mod.rs

```diff
@@ -3140,13 +3141,21 @@ EnumTypeFoldableImpl! {
         (StatementKind::SetDiscriminant) { place, variant_index },
         (StatementKind::StorageLive)(a),
         (StatementKind::StorageDead)(a),
-        (StatementKind::InlineAsm) { asm, outputs, inputs },
+        (StatementKind::InlineAsm)(a),
         (StatementKind::Retag)(kind, place),
         (StatementKind::AscribeUserType)(a, v, b),
         (StatementKind::Nop),
     }
 }
 
+BraceStructTypeFoldableImpl! {
```

Comment:
You can protect against future size regressions by using a static assert like https://github.com/rust-lang/rust/blob/f694222887cf31f51e68927716c25736e62f037f/src/librustc/ty/sty.rs#L2108-L2109